app/views/markets/_marketlist.html.erb

```diff
@@ -0,0 +1,31 @@
+<% @letter_array = [] %>
+
+<% @all_markets.each do |market| %>
+  <% @letter_array << market.market_name[0] %>
+<% end %>
+<% @letters = @letter_array.uniq %>
+
+<ul class = "groupheading">
+  <% @letters.each do |letter| %>
+    <li>
+      <%= link_to letter, "/markets/1/#{letter}", method: :get %>
```

Comment:
Again, this seems like a weird route.  It probably should be changed to something more standardized.